database/changesets/20210510_TerraSinkDetailsTable.xml

```diff
@@ -0,0 +1,27 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext
+                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd
+                        http://www.liquibase.org/xml/ns/dbchangelog
+                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd">
+    <changeSet author="chengche@broadinstitute.org"
+               dbms="postgresql"
+               id="20210510"
+               logicalFilePath="20210510_TerraSinkDetailsTable.xml"
+               runInTransaction="false">
+        <comment>
+            the details table of 'Terra' executor
+        </comment>
+        <sql>
+            CREATE TYPE TerraSinkDetails AS (
+                id                     BIGINT,      -- unique to this table
+                workflow_id            TEXT,        -- id of the Terra workflow
+                entity_name            TEXT,        -- name of the entity
+                updated                TIMESTAMPTZ  -- status update time
+            )
```

Comment:
We also need a `consumed` field since this contains elements of the sink queue.

Reply:
I wasn't sure about this. I can see we peek and pop from source and executor queues, not sure what we do about the sink queue. What do you think `consume` means in sink? Would it be a terminal status of a sample we process?

Reply:
Unique *to* this table or unique *in* this table?
(Foreign or primary key, in other words.)
And so on above …